sql/src/test/java/io/crate/integrationtests/WindowFunctionsIntegrationTest.java

```diff
@@ -41,6 +41,27 @@ public void testMultipleWindowFunctions() {
         assertThat(printedTable(response.rows()), is("1.5| 3\n1.5| 3\n1.5| 3\n"));
     }
 
+    @Test
+    public void testOrderedWindow() {
+        execute("select col1, avg(col1) OVER(ORDER BY col1 NULLS LAST) from unnest([2, 1, 1, 3, 3, null, 4]) order by 1 desc");
+        assertThat(printedTable(response.rows()), is("NULL| 2.3333333333333335\n" +
+                                                     "4| 2.3333333333333335\n" +
+                                                     "3| 2.0\n" +
+                                                     "3| 2.0\n" +
+                                                     "2| 1.3333333333333333\n" +
+                                                     "1| 1.0\n" +
+                                                     "1| 1.0\n"));
+    }
+
+    @Test
+    public void testOrderedWindowWithSingleRowWindows() {
+        execute("select col1, sum(col1) OVER(ORDER BY col1) from unnest([1, 2, 3, 4])");
+        assertThat(printedTable(response.rows()), is("1| 1\n" +
+                                                     "2| 3\n" +
+                                                     "3| 6\n" +
+                                                     "4| 10\n"));
+    }
+
```

Comment:
maybe add an integration test with multiple window functions in one statement?

Reply:
there is one already `testMultipleWindowFunctions`

Reply:
actually, you might be right. adding one that the uses different columns in the different window functions